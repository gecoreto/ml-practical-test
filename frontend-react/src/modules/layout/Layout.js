import {
  BrowserRouter as Router,
  Redirect,
  Route, Switch
} from "react-router-dom";
import Items from '../items/Items';
import Header from './header/Header';

function Layout() {
  return (
    <Router>
      <Header />
      <Switch>
        <Route exact path="/" render={() => (
          <Redirect to="/items" />
        )} />
        <Route path="/items">
          <Items />
        </Route>
      </Switch>
    </Router>
  )
}

export default Layout;
