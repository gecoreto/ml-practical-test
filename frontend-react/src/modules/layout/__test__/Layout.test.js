import React from 'react';
import { shallow } from 'enzyme';
import Layout from '../Layout';

jest.mock('react-redux', () => {
    return {
        useSelector: () => jest.fn(),
        connect: (mapStateToProps, mapDispatchToProps) => (ReactComponent) => ({
            mapStateToProps,
            mapDispatchToProps,
            ReactComponent
        }),
        Provider: ({ children }) => children
    }
})

describe("Layout component", () => {

    const wrapper = shallow(<Layout />);

    test('should test Layout component', () => {
        expect(wrapper).toMatchSnapshot();
    });
});
