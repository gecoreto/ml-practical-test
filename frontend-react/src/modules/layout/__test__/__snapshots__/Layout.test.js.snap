// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout component should test Layout component 1`] = `ShallowWrapper {}`;
